include/wlr/types/wlr_workspace_v1.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_WORKSPACE_V1_H
+#define WLR_TYPES_WLR_WORKSPACE_V1_H
+
+#include <wayland-server-core.h>
+#include <wlr/types/wlr_output.h>
+
+struct wlr_workspace_manager_v1 {
+	struct wl_event_loop *event_loop;
+	struct wl_event_source *idle_source;
+
+	struct wl_global *global;
+	struct wl_list resources; // wl_resource_get_link
```

Comment:
May you please explain what is purpose of resources?

Reply:
A resource is typically a connected client, we just need to keep a list of connected clients so we can send them events.

Reply:
Ok, ty.